pyproject.toml

```diff
@@ -56,3 +55,12 @@ pythonpath = "src/"
 markers = [
     "uncollect_if(*, func): function to uncollect tests from parametrization",
 ]
+
+[tool.ruff]
+target-version = "py38"
+select = ["E", "I", "F", "B", "FA", "UP", "ISC", "PT", "NPY", "Q", "RET", "SIM", "PERF"]
```

Comment:
Will this configuration skip over tests like ['W'](https://beta.ruff.rs/docs/rules/#warning-w)?  Is there any redundancy across suites of tests such that issues that would be flagged on 'W' may be flagged elsewhere as well?  (EDIT: I think black will take care of 'W' rules, perhaps?)

I just tried to see what would pop out if we included *all* rules using this configuration:
```
[tool.ruff]
target-version = "py38"
ignore = []
show-fixes = true
```
but nothing came up with `precommit run --all-files`.  Is that the correct configuration and it just didn't find any issues?  If so, good on us 😄 

I always have the fear of missing out on some automated check, but those can quickly lead to annoyances, so I'm satified that you've probably captured the meaningful ones.

Reply:
> Will this configuration skip over tests like ['W'](https://beta.ruff.rs/docs/rules/#warning-w)? Is there any redundancy across suites of tests such that issues that would be flagged on 'W' may be flagged elsewhere as well? (EDIT: I think black will take care of 'W' rules, perhaps?)

You're right that this doesn't include the `W` ruleset. I think there are potentially redundancies (and even conflicts) between some rulesets, but as you suspected my thought was that these specific cases should all be covered by `black`.

> I just tried to see what would pop out if we included all rules using this configuration: ... but nothing came up with precommit run --all-files. Is that the correct configuration and it just didn't find any issues? If so, good on us 😄

Sadly not! I think `ignore` only excludes selected rules, which defaults to just `E` and `F`. If you try again with `select = ["ALL"]` the results aren't quite as encouraging, although most of the things it catches aren't unexpected at least (docstrings, type annotations, and PEP8 naming are the big ones).

> I always have the fear of missing out on some automated check, but those can quickly lead to annoyances, so I'm satified that you've probably captured the meaningful ones.

It's easy enough to remove rules later if any turn out to be annoying, so I'm open to adding anything that looks useful to you! But it's also easy to add more later, so no need for us to finalize the list now.

Reply:
I think now that I know the trick about how to run with "ALL" (thanks!), I can play around with it (outside of this PR) and see if there are any that I find useful.  It brought up a lot of stuff with `_cca.py` and `_ccora.py` that I will probably adhere to, but like you I'm OK not finalizing the list right now.